### DISCUSSION TITLE
Test is successful but Search doesn't find anything

### DISCUSSION THREAD

**Discussion by mb96851** [facelessuser/Rummage#462] [Q&A] [answered]
This is my first time using Rummage.  In Sublime Text my Find in Files regex searches 77 files and finds 261 matches across 5 files.  In Rummage it searches the same 77 files, skips 0, but doesn't find any matches.  If I use the Regex Tester and copy the contents of one of my files into it, the tester finds the same matches that Sublime Text does.

I have three things checked in **Text search options**:  Search with regex, Create backups, Force ASCII

This is my regex:
`"[ &]*_\n[ &]*"( *(?:INSERT|DELETE|UPDATE|SET|SELECT|INTO|FROM|WHERE|AND|OR|GROUP BY|HAVING|ORDER BY|\( |.*;"|, ))`

I'm trying to clean up, and fix, SQL text strings in a project to all have the same format and I intend to do this in all branches of my code repository to reduce the chance future merges will break things. The above regex is step 1 of 50+ steps, so I'm really hoping I can get Rummage Search Chains to work.

Anyone have any suggestions?

**Comment by facelessuser** [comment]
Please provide a minimal file that we can test. I need a concrete, reproducible example or I cannot begin to guess what you are doing wrong. It is possible that you have your glob pattern configured wrong or any number of issues, but without a reproducible example, I cannot help.

**Comment by facelessuser** [comment]
Now that I mention glob patterns, please take a screenshot showing your file pattern settings, etc. as well.

**Comment by facelessuser** [comment]
I was able to take your regex and construct an example in the regex tester that worked:

<img width="651" alt="Screenshot 2024-07-06 at 7 26 59 AM" src="https://github.com/facelessuser/Rummage/assets/1055125/fe77b0cb-af96-463e-950e-9a1005818df8">

Then I placed the passing example in a file and Rummage found it no problem:

<img width="950" alt="Screenshot 2024-07-06 at 7 28 00 AM" src="https://github.com/facelessuser/Rummage/assets/1055125/bfcd2724-bf9f-4d4e-bd7f-b9eccfe1f42f">

So, I will definitely need more information. I suspect your issue is a user error not a Rummage error, but without knowing exactly what you are doing, I cannot offer further guidance.

**Reply by mb96851** [reply]
Yes, hopefully it's just user error.  :)

Here is an [Example.bas.txt](https://github.com/user-attachments/files/16136012/Example.bas.txt) with one of the routines that needs fixing.
(Github wouldn't let me add Example.bas. Since it is actually at text file, I changed the extension to .txt.)

I pasted the contents of the example into Tester and it finds everything.
![image](https://github.com/facelessuser/Rummage/assets/168605066/44f6e69e-1e60-4260-b7f0-c8d394e39014)

Here is what the search tab looks like before clicking Search.
![image](https://github.com/facelessuser/Rummage/assets/168605066/1ea2a029-d0ef-4ad2-8c03-34923472ddec)
And my no results.
![image](https://github.com/facelessuser/Rummage/assets/168605066/d3001641-83f4-4f7f-a0ac-8e7b0cf7a1e1)

This is the folder I'm trying to search. Everything in the folder, except for the Microsoft Access stub, is a text file.
![image](https://github.com/facelessuser/Rummage/assets/168605066/6910298a-3466-45d3-ae36-060fae6d4a03)

I think i also tweaked the Search Preferences before hand. Here is what they look like.
![image](https://github.com/facelessuser/Rummage/assets/168605066/4cc1f642-af06-427d-b03d-0a5e9b738e8a)
![image](https://github.com/facelessuser/Rummage/assets/168605066/6078ac4c-c039-44cd-9ed4-3a75d01f570f)
![image](https://github.com/facelessuser/Rummage/assets/168605066/aac34e32-a3a6-48ce-b076-17c4b2af9abf)
![image](https://github.com/facelessuser/Rummage/assets/168605066/4c12bff0-f392-49a2-9f2e-32a0730b1c20)

**Comment by facelessuser** [comment]
Just to be clear, the directory you are showing me in Windows file explorer does not match the path in rummage, you noticed this right?

**Reply by facelessuser** [reply]
I assume this may be because your results were applied to the test folder with a single file vs. the folder you are targeting in real life? Just want to make sure this wasn't all related to something as simple as scanning the wrong directory.

**Reply by mb96851** [reply]
Oops.  Here is the actual source folder I'm trying to search. 
![image](https://github.com/facelessuser/Rummage/assets/168605066/101d08d2-fd1c-488f-8ce6-c0c27aa2197a)

And Sublime Text finding the test. 
![image](https://github.com/facelessuser/Rummage/assets/168605066/6ba66484-89f3-4b97-a951-bcc3d4cc6688)

**Reply by facelessuser** [reply]
Thanks. I'll investigate and report back with my findings.

**Reply by mb96851** [reply]
Yeah always good to double & triple check. I have three clones updated to 3 different branches. I'm actually hoping to make these changes to all of them at the same time.

**Reply by facelessuser** [reply]
One more question, are there symlinks involved? Or whatever the windows equivalent is called.

**Reply by facelessuser** [reply]
@mb96851 Also, can you tell me what line endings the file uses? You can enable `"show_line_endings": true,` in Sublime and it will show you the line endings in the status bar. Or go to `view -> line endings` and tell what is checked. If it is Windows, then `\n` is probably not going to cut it. Sublime will normalize line endings in the editor view, but when searching the raw file, windows line mode will use `\r\n`. Let me know your findings before I continue any further.

**Reply by facelessuser** [reply, marked as answer]
@mb96851 Yeah, the file you gave me had Windows new lines. Since you have `backrefs` enabled, just use `\R` which will file line endings of `\n`, `\r`, or `\r\n`. Either that or use explicit `\r\n`. Once I updated the pattern to use `\R` instead or `\n`, it all worked fine.

**Comment by mb96851** [comment]
Thanks for figuring it out!
